src/display.cpp

```diff
@@ -233,7 +258,12 @@ display::display(const display_context * dc, std::weak_ptr<wb::manager> wb, repo
 
 	set_idle_anim_rate(preferences::idle_anim_rate());
 
-	zoom_index_ = std::distance(zoom_levels.begin(), std::find(zoom_levels.begin(), zoom_levels.end(), zoom_));
+	if(preferences::tile_size() > 0) {
+		zoom_ = preferences::tile_size();
+		validate_zoom_level_and_set_index(zoom_, zoom_index_);
```

Comment:
Unless I'm misreading this, if the preferences file contains an invalid value, then validate_zoom_level_and_set_index will return false, but `zoom_` will remain set to the value in the preferences file and `zoom_index_` and `zoom_` will remain out of sync. I suggest to read the preferences into a temporary variable and only set `zoom_` and `zoom_index_` after validating the preferences value.

(Actually, I'm not sure why we have both of them. Having two things that need to be kept in sync is, generally, a bug waiting to happen. But refactoring that would be beyond the scope of this PR.)

Reply:
I think the dual `zoom_`/`zoom_index_` thing is giving me head-aches. I did consider making attempt to refactor this but it looks like `zoom_` is also being used to calculate the hex/tile sizes.

~I decided to switch to recording `zoom_factor` instead - both `zoom_` and `zoom_index_` can be derived from this single value rather than trying to co-ordinate these two values. I think this change also resolves your early question about whether we're recording enough information.~

I belatedly realised that while `game_config.cfg` defines `zoom_levels` as an array of zoom factors, but when it gets loaded in `game_config::load_config()` it converts it so that `game_config::zoom_levels` becomes a vector of tile sizes. So I've reverted to recording the `tile_size` which means I'm back to the `zoom_` and `zoom_index_` pairing...

Now that I look at it again, I think there's no real purpose to `zoom_index_` (that I can see). Maybe I should see if I can get rid of it altogether...

Reply:
Oh I see... `zoom_index_` seems to be there purely for when using the zoom in/out hot-keys, the index can just be incremented or decremented and the corresponding value in the `zoom_levels` vector can be used easily. I know, I'm slow at understanding this...